x-pack/plugins/task_manager/server/task_claimers/strategy_update_by_query.test.ts

```diff
@@ -706,36 +815,50 @@ if (doc['task.runAt'].size()!=0) {
         },
       });
 
-      async function getUpdateByQueryScriptParams() {
-        return (
-          await getAllAsPromise(
-            taskClaiming.claimAvailableTasksIfCapacityIsAvailable({
-              claimOwnershipUntil: new Date(),
-            })
-          )
-        ).map(
-          (result, index) =>
-            (
-              store.updateByQuery.mock.calls[index][0] as {
-                query: MustNotCondition;
-                size: number;
-                sort: string | string[];
-                script: {
-                  params: {
-                    [claimableTaskTypes: string]: string[];
-                  };
+      // first cycle
+      await taskClaiming.claimAvailableTasksIfCapacityIsAvailable({
+        claimOwnershipUntil: new Date(),
+      });
+      expect(store.updateByQuery).toHaveBeenCalledTimes(4);
+      const firstCycle = store.updateByQuery.mock.calls.map(
+        (call) =>
+          (
+            call[0] as {
+              query: MustNotCondition;
+              size: number;
+              sort: string | string[];
+              script: {
+                params: {
+                  [claimableTaskTypes: string]: string[];
                 };
-              }
-            ).script.params.claimableTaskTypes
-        );
-      }
+              };
+            }
+          ).script.params.claimableTaskTypes
+      );
 
-      const firstCycle = await getUpdateByQueryScriptParams();
       store.updateByQuery.mockClear();
-      const secondCycle = await getUpdateByQueryScriptParams();
 
-      expect(firstCycle.length).toEqual(4);
-      expect(secondCycle.length).toEqual(4);
+      // second cycle
+      await taskClaiming.claimAvailableTasksIfCapacityIsAvailable({
+        claimOwnershipUntil: new Date(),
+      });
+      expect(store.updateByQuery).toHaveBeenCalledTimes(4);
+      const secondCycle = store.updateByQuery.mock.calls.map(
+        (call) =>
+          (
+            call[0] as {
```

Comment:
We can extract this type as the following, and reuse it in the `firstCycle` setting above this as well.

```
      interface UBQParams {
        script: {
          params: {
            [claimableTaskTypes: string]: string[];
          };
        };
      }

      // ...

      const secondCycle = store.updateByQuery.mock.calls.map(
        (call) => (call[0] as UBQParams).script.params.claimableTaskTypes
      );
```

Reply:
Updated in https://github.com/elastic/kibana/pull/196196/commits/5da0cf8afcd192f5fcec18e8aa5bfc7afe2442c1